hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ReplaceCommitValidateUtil.java

```diff
@@ -0,0 +1,71 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.action.cluster;
+
+import org.apache.hudi.common.model.HoodieReplaceCommitMetadata;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.JsonUtils;
+import org.apache.hudi.exception.HoodieException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.apache.hudi.common.util.FileIOUtils.LOG;
+
+public class ReplaceCommitValidateUtil {
+  public static final String REPLACE_COMMIT_FILE_IDS = "replaceCommitFileIds";
+  public static void validateReplaceCommit(HoodieTableMetaClient metaClient) {
+    metaClient.reloadActiveTimeline();
+    Set<String> replaceFileids = new HashSet<>();
+
+    // Verify pending and completed replace commit
+    Stream.concat(metaClient.getActiveTimeline().getCompletedReplaceTimeline().getInstants().stream(),
+        metaClient.getActiveTimeline().filterInflights().filterPendingReplaceTimeline().getInstants().stream()).map(instant -> {
+          try {
+            HoodieReplaceCommitMetadata replaceCommitMetadata =
+                HoodieReplaceCommitMetadata.fromBytes(metaClient.getActiveTimeline().getInstantDetails(instant).get(),
+                HoodieReplaceCommitMetadata.class);
+            if (!instant.isCompleted()) {
+              return JsonUtils.getObjectMapper().readValue(replaceCommitMetadata.getExtraMetadata().getOrDefault(REPLACE_COMMIT_FILE_IDS, "non-existent key"), String[].class);
+            } else {
+              return replaceCommitMetadata.getPartitionToReplaceFileIds().values().stream()
+                  .flatMap(List::stream)
+                  .toArray(String[]::new);
+            }
+          } catch (IOException e) {
+            // If the key does not exist or there is a JSON parsing error, LOG reports an error and ignores it.
+            LOG.error("Error when reading replace commit meta", e);
+            return null;
+          }
+        }).filter(Objects::nonNull)
+        .forEach(fileIdArray -> {
+          Arrays.stream(fileIdArray)
+              .filter(fileId -> !replaceFileids.add(fileId))
+              .findFirst()
+              .ifPresent(s -> {
+                throw new HoodieException("Replace commit involves duplicate file id!");
```

Comment:
Can make the exception more clear

Reply:
Reply “I think we better throw the exception here instead of return null”：
I made the decision based on the following consideration: there may be some ways, such as constructing tests that do not create an inflight status timeline through normal execution processes. My additional validation information will not be added here, but if such a situation occurs during execution, an error will be reported here due to parsing exceptions. I do not want the additional validation procedures introduced here to affect the original logic. Without my validation information, the replace function will not be executed properly. Therefore, I only report the error through logging, but do not throw an exception. Do you think this makes sense?

Reply:
I will revise this description